internal/namespace/diff.go

```
@@ -149,17 +178,47 @@ func DiffNamespaces(existing *core.NamespaceDefinition, updated *core.NamespaceD
 		})
 	}
 
+	for _, removed := range strset.Difference(existingPermNames, updatedPermNames).List() {
+		deltas = append(deltas, Delta{
+			Type:         RemovedPermission,
+			RelationName: removed,
+		})
+	}
+
+	for _, added := range strset.Difference(updatedPermNames, existingPermNames).List() {
+		deltas = append(deltas, Delta{
+			Type:         AddedPermission,
+			RelationName: added,
+		})
+	}
+
+	for _, shared := range strset.Intersection(existingPermNames, updatedPermNames).List() {
+		existingPerm := existingPerms[shared]
+		updatedPerm := updatedPerms[shared]
+
+		// Compare implementations.
+		m := &jsonpb.Marshaler{}
+		existingRewriteJSON, _ := m.MarshalToString(existingPerm.UsersetRewrite)
+		updatedRewriteJSON, _ := m.MarshalToString(updatedPerm.UsersetRewrite)
```

This might need to panic instead of ignoring the error